### PR TITLE
New 'custom metadata table' lightning web component

### PR DESCRIPTION
**Pull request by jongpie**
The repo currently supports creating, updating & deploying CMDT records via backend automations (Apex and Flow) - this PR adds a new lightning web component `custom-metadata-table` that provides a way for users to create, update & deploy CMDT records through custom UIs (lwc, aura components and screen Flows).

### Using `custom-metadata-table` component
The component can be used within screen Flows or within other lightning web/aura components. It requires 3 parameters:
1. `records` - an instance of `List<SObject>` in Apex, or a 'record collection' within Flow, containing the CMDT records to display within the table. All of the `SObject` records must be of the same `SObject Type`. You can create or query the list of CMDT records - see README for details on how to create CMDT records in Apex and All CMDT records must have both fields `DeveloperName` and `MasterLabel` populated in order to successfully deploy the new records.
2. `fieldsToDisplay` - a comma-separated list of field API names that should be shown in the table. When the paremeter `enableEditing == true`, all fields, except `DeveloperName` and lookup fields, will be editable within the table. This is done via the `lightning-datatable`'s inline-editing functionality
3. `enableEditing` - Boolean value to control if CMDT records are displayed in read-only mode (`enableEditing == false`) or with inline-editing enabled (`enableEditing == true`)

### Flow Example
These screenshots show an example of using the component within Flow. In this first image, Flow's `Get Record` feature is used to query CMDT records in the CMDT object `CustomMetadataDeployTest__mdt` - but everything is done dynamically, so you can use any custom metadata type (CMDT) object.

image

Once you've queried or created a record collection containing CMDT records, you can add a screen that uses the `Custom Metadata Table' component.

image

### The End Result
Once the table is configured, it displays the CMDT records in a `lighting-datatable` component

image

When users edit 1 or more fields, the standard `Save` and `Cancel` buttons are automatically displayed on the `lightning-datable` component:
- When a user clicks `Cancel`, any unsaved changes to the CMDT records are discarded, and all field values are reverted to their original value.
- When a user clicks `Save`, a loading-spinner is displayed to prevent any additional changes while the CMDT records are deployed asynchronously to the current org. Once the deployment finishes (either by successfully deploying the CMDT records or by failing due to an error), then the component removes the loading-spinner-icon, and users can resume interacting with the component.

### Possible Future Enhancements
This PR provides the core functionality for a CMDT-based datatable, but there are definitely some features currently missing - hopefully these features (and others) will be added in future releases
- Picklist fields are not currently properly rendered as a picklist - this PR's code renders them as a text field, which works, but this needs to be addressed in a future
- `New` button/action to create a new CMDT record from within the table
- Column sorting functionality
- Ability to navigate to a CMDT record's 'View' page (under Setup)

### CODE REVIEW

File: force-app/main/custom-metadata-saver/classes/CustomMetadataTableController.cls [hunk 1]
```diff
@@ -0,0 +1,128 @@
+//----------------------------------------------------------------------------------------------------//
+// This file is part of the Custom Metadata Saver project, released under the MIT License.            //
+// See LICENSE file or go to https://github.com/jongpie/CustomMetadataSaver for full license details. //
+//----------------------------------------------------------------------------------------------------//
+
+public without sharing class CustomMetadataTableController {
+    private static String BASE_URL = System.Url.getOrgDomainUrl().toExternalForm() + '/services/data/v49.0';
```
Comment (jamessimone) [inline force-app/main/custom-metadata-saver/classes/CustomMetadataTableController.cls:7]: you've got my attention with that API version - any particular reason for `49` ?

File: force-app/main/custom-metadata-saver/lwc/customMetadataTable/customMetadataTable.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    isDeploying = false;
+    deploymentStatus;
+    deploymentMessage;
+    _deploymentId;
+    _resolvedDeploymentStatuses = ['Succeeded', 'Failed', 'Aborted'];
+
+    connectedCallback() {
+        // For SObject and List<SObject> variables in Flow, Flow Builder automatically asks
+        // admins for the SObject Type - but this isn't exposed to lwc without using custom property editors (CPE),
+        // which are terrible - 2/10 stars, would not recommend to a friend
```
Comment (jamessimone) [inline force-app/main/custom-metadata-saver/lwc/customMetadataTable/customMetadataTable.js:44]: this comment, on the other hand, is 10/10

File: force-app/main/custom-metadata-saver/classes/CustomMetadataTableController.cls (hunk 1 above)
Comment (jongpie) [reply force-app/main/custom-metadata-saver/classes/CustomMetadataTableController.cls:7]: I think this is a result of some good ole copypasta - I copied part of this from my ApexCallouts repo & forgot to update the API version, I'll update it!

File: force-app/main/custom-metadata-saver/lwc/customMetadataTable/customMetadataTable.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+                if (deploymentStatusResponse.deployResult.details.componentFailures[0]) {
+                    this.deploymentMessage = 'Error: ' + deploymentStatusResponse.deployResult.details.componentFailures[0].problem;
+                } else {
+                    this.deploymentMessage = null;
+                }
+            }
+        }
+
+        const statusPromise = new Promise(resolve => {
```
Comment (jamessimone) [inline force-app/main/custom-metadata-saver/lwc/customMetadataTable/customMetadataTable.js:235]: classic

File: force-app/main/custom-metadata-saver/classes/CustomMetadataTableController.cls (hunk 1 above)
Comment (jongpie) [reply force-app/main/custom-metadata-saver/classes/CustomMetadataTableController.cls]: Fixed! I've bumped it to `v52.0` instead of `v49.0`

**Comment by jamessimone** [comment]
🎉!

**Pull request merged**